ios/Classes/ComWilliamrijksenOnesignalModule.m

```diff
@@ -13,93 +13,120 @@
+        body = [payload.body copy];
+    }
 
-        NSDictionary* notificationData = @{
-                                            @"title": title,
-                                            @"body": body,
-                                            @"additionalData": additionalData
-                                        };
-        [self fireEvent:@"notificationReceived" withObject:notificationData];
+    if (payload.additionalData) {
+        additionalData = payload.additionalData;
     }
-};
-    
-- (void) actionHandler:(OSNotificationOpenedResult *)result {
-    if ([self _hasListeners:@"notificationOpened"]) {
-        OSNotificationPayload* payload = result.notification.payload;
 
-        NSString* title = @"";
-        NSString* body = @"";
-        NSDictionary* additionalData = [[NSDictionary alloc] init];
+    NSDictionary* notificationData = @{
+                                        @"title": title,
+                                        @"body": body,
+                                        @"additionalData": additionalData
+                                    };
+    [self tryToPostNotification:notificationData withNotificationName:TiNotificationReceived];
+}
 
-        if(payload.title) {
-            title = payload.title;
-        }
++ (void) actionHandler:(OSNotificationOpenedResult *)result {
+    OSNotificationPayload* payload = result.notification.payload;
 
-        if (payload.body) {
-            body = [payload.body copy];
-        }
+    NSString* title = @"";
+    NSString* body = @"";
+    NSDictionary* additionalData = [[NSDictionary alloc] init];
 
-        if (payload.additionalData) {
-            additionalData = payload.additionalData;
-        }
+    if(payload.title) {
+        title = payload.title;
+    }
 
-        NSDictionary* notificationData = @{
-                                       @"title": title,
-                                       @"body": body,
-                                       @"additionalData": additionalData};
-        [self fireEvent:@"notificationOpened" withObject:notificationData];
+    if (payload.body) {
+        body = [payload.body copy];
     }
+
+    if (payload.additionalData) {
+        additionalData = payload.additionalData;
+    }
+
+    NSDictionary* notificationData = @{
+                                    @"title": title,
+                                    @"body": body,
+                                    @"additionalData": additionalData};
+    [self tryToPostNotification:notificationData withNotificationName:TiNotificationOpened];
 }
 
 - (void)startup
 {
-   [super startup];
-   instance = self;
-   [OneSignal promptForPushNotificationsWithUserResponse:^(BOOL accepted) {
-        NSLog(@"User accepted notifications: %d", accepted);
-   }];
+    [super startup];
+    NSLog(@"[INFO] started %@", self);
+    _instance = self;
+//    [self performSelector:@selector(handleQueuedBootEvents) withObject:self afterDelay:10.0];
```

Comment:
Oh yes, my bad. It should be `launch`. When it's `launch`, I see `launch is being called, but the fireEvent is not working. When I call `launch like this (https://github.com/williamrijksen/com.williamrijksen.onesignal/pull/38#issuecomment-314575600) it's working properly. Do you know why that's failing?